File: 3379-transformed-array/3379-transformed-array.cpp
class Solution {
public:
    vector<int> constructTransformedArray(vector<int>& nums) {
        vector<int>ans;
        int n = nums.size();
        for(int i=0; i<n; i++)
        {
            int pos = ((i + nums[i]) % n + n) % n;
            ans.push_back(nums[pos]);
        }
        return ans;
    }
};

File: 3379-transformed-array/3379-transformed-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var constructTransformedArray = function(nums) {
    let n=nums.length;
    let ans = [];
    for(i=0; i<n; i++)
    {
        let j = ((nums[i] + i) % n + n )% n;
        ans[i]=nums[j];
    }
    return ans;
};